### PR TITLE
Clean up Compat modules

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Many of the compatibility modules are no longer needed because they are either empty or contain the same code for all supported versions of Chapel. This PR removes the unnecessary compat modules and updates the codebase accordingly.

With this change, we only have the following four compat modules:
* ArkoudaAryUtilCompat (can be removed when 1.32 support is dropped)
* AkoudaIOCompat
* ArkoudaRandomCompat
* ArkoudaSortCompat

- [x] tests passing w/ gasnet build

### CODE REVIEW

File: src/StatsMsg.chpl [hunk 1]
```diff
@@ -343,7 +343,7 @@ module StatsMsg {
             var eOut = st.addEntry(rname, outShape, t);
 
             const DD = domOffAxis(eIn.a.domain, axis);
-            forall idx in DD {
```
Comment (stress-tess) [inline src/StatsMsg.chpl:346]: just want to make sure this bit is intentional since it seems unrelated to the compat modules

Comment (stress-tess) [reply src/StatsMsg.chpl:346]: I didn't highlight all of it apparently but the `forall` is being turned into a `for`

Comment (jeremiah-corrado) [reply src/StatsMsg.chpl:346]: yeah, that change fixed an error I was seeing for nd-array builds. Just happened to notice it while working on this.
